stock_lot_scrap/models/stock_production_lot.py

```diff
@@ -72,7 +71,18 @@ def action_scrap_lot(self):
             )
         scrap_obj = self.env["stock.scrap"]
         scraps = scrap_obj.browse()
-        scrap_location_id = self.env.ref("stock.stock_location_scrapped").id
+        # The data model has disappeared and right now is a function who creates the
+        # scrap location. That's why we have to search by company_id and scrap_location.
+        scrap_location_id = (
+            self.env["stock.location"]
+            .search(
+                [
+                    ("company_id", "=", self.company_id.id),
+                    ("scrap_location", "=", True,),
+                ]
+            )
+            .id
+        )
```

Comment:
I think this search may cause problems in the future, you can make it more resilient by limiting the search results to 1:
```suggestion
        # The data model has disappeared and right now is a function who creates the
        # scrap location. That's why we have to search by company_id and scrap_location.
        scrap_location_id = (
            self.env["stock.location"]
            .search(
                [
                    ("company_id", "=", self.company_id.id),
                    ("scrap_location", "=", True,),
                ],
                limit=1,
            )
            .id
        )